src/findings_data_import.py

```diff
@@ -49,12 +53,372 @@ def move_processed_file(s3_client, bucket, folder, filename):
+
+        # work with v1 and v2. If has NCATS ID OR findings the document is probably OK
+        if (
+            "RVA ID" not in finding.keys()
+            or "NCATS ID" not in finding.keys()
+            or "Severity" not in finding.keys()
+        ):
+            logging.warning(
+                'Skipping record %d. Missing "RVA ID", "NCATS ID", or "Severity" field.',
+                index,
+            )
+            continue
+
+        # Get RVA ID in format DDDD([.-]D+) from the end of the "RVA ID" field.
+        rvaId = re.search(r"(\d{4})(?:[.-](\d+))?$", finding["RVA ID"])
+        if rvaId:
+            rID = f"RV{rvaId.group(1)}"
+            if rvaId.group(2) is not None:
+                rID += f".{rvaId.group(2)}"
+            finding["RVA ID"] = rID
+        else:
+            logging.warning(
+                'Skipping record %d: Unable to extract valid RVA ID from "%s"',
+                index,
+                finding["RVA ID"],
+            )
+            continue
+        # flag this as V1 so update knows how to handle it, and its clear to folks viewing the data downstream
+        finding["Schema"] = V1_SCHEMA
+        valid_findings.append(finding)
+
+    return valid_findings
+
+
+def validate_v2_findings(findings_data: dict):
+    """Validate a list of V1 findings, discarding invalid entries (such as those with no severity which are not explicitly 'findings').
+
+    Parameters
+    ----------
+    findings_data : dict
+        The findings dictionary pulled from a findings JSON file (v2-style)
+
+    field_map_dict: dict
+        The dictionary of replacement rules for field names in findings_data (used only in V1 style processing)
+
+    Returns
+    -------
+    list : A list of findings objects from findings_data that pass validation
+    """
+    try:
+        if not findings_data or not isinstance(findings_data, dict):
+            logging.warning("Received an empty or invalid finding object, skipping.")
+            raise ValueError("Received an empty or invalid object.")
+
+        # work with v1 and v2. If has NCATS ID OR findings the document is probably OK
+        if "id" not in findings_data.keys() or "findings" not in findings_data.keys():
+            logging.warning('Skipping record. Missing "id" or "findings" field.')
+            raise ValueError("Missing id or findings field in v2 findings object")
+
+        # Get RVA ID in format DDDD([.-]D+) from the end of the "RVA ID" field.
+        rvaId = re.search(r"(\d{4})(?:[.-](\d+))?$", findings_data["id"])
+        if rvaId:
+            rID = f"RV{rvaId.group(1)}"
+            if rvaId.group(2) is not None:
+                rID += f".{rvaId.group(2)}"
+            findings_data["id"] = rID
+        else:
+            logging.warning(
+                "Skipping record. Unable to extract valid RVA ID from %s",
+                findings_data["id"],
+            )
+            raise ValueError("Unable to parse RVA id from finding object")
```

Comment:
We should not be modifying anything coming out of a new v2 JSON.

```suggestion
```

Reply:
Why is the original field modified? I'd love to see the requirements doc -- this is present in the existing logic and I didn't remove it because I don't see any reason to change that specific behavior that is applicable to both schemas.

Looking back in the history, you were involved in the original code that does this, can you explain why the change happens at all, and why it would be different now?

Reply:
Unfortunately there was no original requirements doc to my knowledge and I came in to help the contractor assigned the task toward the end. The `RVA ID` in the v1 schema is changed because (at the time, I don't know about now) the provided assessment ID could include a "phase" e.g. `RV0801.01`. This flattened any such IDs to a single, consistent ID e.g. `RV0801`. This works for the v1 data because all findings are individual documents. It is safe to overwrite a given single finding as it would apply to the same assessment. With v2 even if they were still using the "phase" in the `id` it would **not** be safe to overwrite a document because it could result in the loss of data.

Reply:
Makes sense, but should probably check with RVA so they know one way or the other, they may not have considered this at all.

Reply:
@Yablargo Due to the increased chance of overwriting data with the entire file being saved as one entry, please proceed with the change to _not_ modify the `RVA ID` field when processing v2 data.  I'll be sure to coordinate with RVA and inform them of the change and the reasoning behind it.

Reply:
@jmorrowomni  I've removed the RVA ID format parsing/validation and it will now take any RVA ID passed in.

Reply:
Only the person who originally created the conversation (or someone that person directed) should be resolving conversations.